src/otx/algo/detection/yolov9.py

```diff
@@ -0,0 +1,196 @@
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+    from typing_extensions import Self
+
+    from otx.core.metrics import MetricCallable
+    from otx.core.schedulers import LRSchedulerListCallable
+    from otx.core.types.label import LabelInfoTypes
+
+
+PRETRAINED_WEIGHTS: dict[str, str] = {
+    # "yolov9_t": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-t.pt", # not supported yet
+    "yolov9_s": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-s.pt",
+    "yolov9_m": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-m.pt",
+    "yolov9_c": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-c.pt",
+    # "yolov9_e": "https://github.com/WongKinYiu/YOLO/releases/download/v1.0-alpha/v9-e.pt", # not supported yet
+}
+
+
+def _load_from_state_dict_for_yolov9(
+    self: SingleStageDetector,
+    state_dict: dict,
+    prefix: str,
+    local_metadata: dict,
+    strict: bool,
+    missing_keys: list[str] | str,
+    unexpected_keys: list[str] | str,
+    error_msgs: list[str] | str,
+) -> None:
+    backbone_len: int = len(self.backbone.module)
+    neck_len: int = len(self.neck.module)  # type: ignore[union-attr]
+    new_state_dict_keys: dict[str, str] = {}
+    for key in state_dict:
+        match = re.match(r"^(\d+)\.(.*)$", key)
+        if match:
+            orig_idx = int(match.group(1))
+            rest_key = match.group(2)
+
+            if orig_idx < backbone_len:
+                new_key = f"backbone.module.{orig_idx}.{rest_key}"
+            elif orig_idx < backbone_len + neck_len:
+                new_idx = orig_idx - backbone_len
+                new_key = f"neck.module.{new_idx}.{rest_key}"
+            else:  # for bbox_head
+                new_idx = orig_idx - backbone_len - neck_len
+                new_key = f"bbox_head.module.{new_idx}.{rest_key}"
+            new_state_dict_keys[key] = new_key
+
+    for old_key, new_key in new_state_dict_keys.items():
+        value = state_dict.pop(old_key)
+        state_dict[new_key] = value
+
+    super(SingleStageDetector, self)._load_from_state_dict(
+        state_dict,
+        prefix,
+        local_metadata,
+        strict,
+        missing_keys,
+        unexpected_keys,
+        error_msgs,
+    )
+
+
+class YOLOv9(ExplainableOTXDetModel):
+    """OTX Detection model class for YOLOv9.
+
+    Default input size per model:
+        - yolov9_t : (640, 640) # not supported yet
+        - yolov9_s : (640, 640)
+        - yolov9_m : (640, 640)
+        - yolov9_c : (640, 640)
+        - yolov9_e : (640, 640) # not supported yet
+    """
+
+    input_size_multiplier = 32  # TODO (sungchul): need to check
+    mean: tuple[float, float, float] = (0.0, 0.0, 0.0)
+    std: tuple[float, float, float] = (255.0, 255.0, 255.0)
+
+    def __init__(
+        self,
+        model_name: Literal["yolov9_t", "yolov9_s", "yolov9_m", "yolov9_c", "yolov9_e"],
```

Comment:
```suggestion
        model_name: Literal["yolov9_s", "yolov9_m", "yolov9_c"],
```
How about removing them for now and updating when they'll become available.

Reply:
Done at 73b7672